server/svix-server/src/worker.rs

```diff
@@ -546,6 +574,43 @@ fn bytes_to_string(bytes: bytes::Bytes) -> String {
     }
 }
 
+async fn get_retry_schedule_override(
+    db: &DbConn,
+    cache: &Cache,
+    org_id: &OrganizationId,
+    event_type_name: &EventTypeName,
+) -> Result<Option<RetrySchedule>> {
+    let cache_key = RetryScheduleCacheKey::new(org_id, event_type_name);
+
+    if let Some(retry_schedule) = cache
+        .get::<RetryScheduleCacheValue>(&cache_key)
+        .await
+        .map_err(|e| Error::from(e.to_string()))?
+    {
+        return Ok(retry_schedule.retry_schedule);
+    };
+
+    let retry_schedule =
+        eventtype::Entity::secure_find_by_name(org_id.to_owned(), event_type_name.to_owned())
+            .one(db)
+            .await?
+            .unwrap()
+            .retry_schedule;
+
+    cache
+        .set(
+            &cache_key,
+            &RetryScheduleCacheValue {
+                retry_schedule: retry_schedule.clone(),
```

Comment:
Yes -- I also return `retry_schedule` at the end of the function.

Reply:
I'm pretty sure you're incorrect, you can just do:
```rust
let retry_schedule_value = RetryScheduleCacheValue { retry_schedule };
cach.set(...., &retry_schedule_value);
...
Ok(retry_schedule_value.retry_schedule).
```

Reply:
Unfortunately no, cannot move out of retry_schedule_value.retry_schedule which is behind a shared reference.

Reply:
I'm pretty sure you have a small error there, but ANYHOW, the only reason why we even need to think about this and deal with this is because we have the wrong type in the first place...
It's trivial to do the serialize and desrialize for duration.
```rust
            let buf: Vec<u64> = Vec::deserialize(deserializer)?;
            Ok(buf.into_iter().map(|x| Duration::new(x, 0)).collect())
```
This is roughly the deserialize function. I'm sure the serialize one is even simpler. We already have it (kinda, though with hacks because of legacy stuff) in cfg.rs, this will be solved if the typing was just correct.

Reply:
OK, I mixed up two topics it seems, ignore my previous message. :)

Though you can definitely make it work, just put the `cache.set` inside a block of curlys `{}`

Reply:
I'll take another look at the typing/serialization regardless

Reply:
It's really easy. You can take a look at types.rs for inspiration, we do it for a lot of types.